source/cell/cell.cpp

```diff
@@ -767,9 +814,115 @@ bool cell::has_format() const
 
 void cell::format(const class format new_format)
 {
+    // Check if format belongs to a different workbook (dangling pointer risk)
+    if (!workbook().owns_format(new_format))
+    {
+        copy_format_from_other_workbook(new_format);
+        return;
+    }
+
+    // Same workbook: direct assignment (original behavior)
     d_->format_ = new_format.d_;
 }
 
+void cell::copy_format_from_other_workbook(const class format &source_format)
+{
+    // Option B: Automatically clone the format to this workbook's stylesheet
+    // Start with a blank format from our workbook
+    auto cloned_format = workbook().create_format(true);
```

Comment:
I don't think you want to change the default format here (i.e. pass true as first argument). This sets the default format of the workbook (although the concept of default format is not fully implemented in xlnt).